File: csharp/src/Vista.SDK/ILocalId.cs
namespace Vista.SDK;

public interface ILocalId
{
    VisVersion VisVersion { get; }

    bool VerboseMode { get; }

    GmodPath PrimaryItem { get; }

    GmodPath? SecondaryItem { get; }

    bool HasCustomTag { get; }

    IReadOnlyList<MetadataTag> MetadataTags { get; }

    string ToString();
}

public interface ILocalId<T> : ILocalId, IEquatable<T>
    where T : ILocalId<T>
{
#if NET7_0_OR_GREATER
    static abstract T Parse(string localIdStr);
    static abstract bool TryParse(string localIdStr, out ParsingErrors errors, out T? localId);
#endif
}

File: js/lib/ILocalId.ts
import { GmodPath } from "./GmodPath";
import { ILocalIdBuilder } from "./ILocalIdBuilder";
import { MetadataTag } from "./MetadataTag";
import { VisVersion } from "./VisVersion";

export interface ILocalId {
    visVersion: VisVersion;
    verboseMode: boolean;
    primaryItem: GmodPath;
    secondaryItem: GmodPath | undefined;
    hasCustomTag: boolean;
    metadataTags: MetadataTag[];

    builder: ILocalIdBuilder;

    toString(): string;
}

export interface ILocalIdGeneric<T extends ILocalIdGeneric<T>>
    extends ILocalId {}
